lib/fabrik/DirectorInstance.js

```diff
@@ -76,6 +76,12 @@ class DirectorInstance extends BaseInstance {
           return this.createSecurityGroup(operation.space_guid);
         case 'update':
           return this.ensureSecurityGroupExists();
+        case 'delete':
+          if (operation.state === 'done') {
+            var deploymentName = this.parseTaskid(operation.task_id, 1);
+            logger.info(`Deleting cache entry for ${deploymentName}`);
+            this.manager.director.deleteCacheEntry(deploymentName);
```

Comment:
Not required. Whats the harm in having one additional entry. Cache anyways gets refresh on restart. As explained, components outside of BoshDirectorClient must not know anything about cache.